src/lxc/lxccontainer.c

```diff
@@ -4840,7 +4840,66 @@ static int do_lxcapi_mount(struct lxc_container *c,
 }
 
 WRAP_API_6(int, lxcapi_mount, const char *, const char *, const char *,
-	   unsigned long, const void *, struct lxc_mount*)
+			unsigned long, const void *, struct lxc_mount*)
+
+static int do_lxcapi_umount2(struct lxc_container *c, const char *target,
+			  unsigned long mountflags, struct lxc_mount *mnt) {
+	pid_t pid, init_pid;
+	int ret = -1;
+
+	if (!c || !c->lxc_conf) {
+		ERROR("Container or configuration is NULL");
+		return -EINVAL;
+	}
+
+	/* Do the fork */
+	pid = fork();
+	if (pid < 0) {
+		SYSERROR("Could not fork");
+		return -1;
+	}
+	
+	if (pid == 0) {
+		init_pid = do_lxcapi_init_pid(c);
+		if (init_pid < 0) {
+			ERROR("Failed to obtain container's init pid");
+			_exit(EXIT_FAILURE);
+		}
+	
+		/* Enter the container namespaces */
+		if (am_guest_unpriv() && !lxc_list_empty(&c->lxc_conf->id_map)) {
+			/* if fully unprivileged */
```

Comment:
I think you can remove the `am_guest_unpriv()` and simply rely on the `lxc_list_empty` check. :)

Reply:
You can also remove the comment.

Reply:
Why can we do it here? I mean, remove the `am_guest_unpriv()` call O.o

Reply:
Oh, you're interested in the fully-unprivileged use-case. Hm, I guess we can leave it in then. :)